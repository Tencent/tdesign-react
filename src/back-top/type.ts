/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode } from '../common';

export interface TdBackTopProps {
  /**
   * 是否绝对定位固定到屏幕右下方
   * @default true
   */
  fixed?: boolean;
  /**
   * 图标
   * @default 'backtop'
   */
  icon?: TNode;
  /**
   * 定位滚动到指定对象
   * @default () => window
   */
  target?: () => HTMLElement;
  /**
   * 文案
   * @default ''
   */
  text?: string;
  /**
   * 预设的样式类型
   * @default round
   */
  theme?: 'default' | 'primary' | 'dark';

  /**
   *  组件的形状
   */
  shape?: 'square' | 'circle';

  /**
   * 组件的大小
   */
  size?: 'small' | 'medium';
}
