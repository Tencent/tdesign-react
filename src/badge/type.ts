/* eslint-disable */

/**
 * 该文件为脚本自动生成文件，请勿随意修改。如需修改请联系 PMC
 * */

import { TNode } from '../common';

export interface TdBadgeProps {
  /**
   * 徽标内容，同 content
   */
  children?: TNode;
  /**
   * 颜色
   * @default ''
   */
  color?: string;
  /**
   * 徽标内容
   */
  content?: TNode;
  /**
   * 徽标右上角内容。可以是数字，也可以是文字。如：'new'/3/99+
   * @default 0
   */
  count?: TNode;
  /**
   * 是否为红点
   * @default false
   */
  dot?: boolean;
  /**
   * 封顶的数字值
   * @default 99
   */
  maxCount?: number;
  /**
   * 设置状态点的位置偏移，示例：[-10, 20] 或 ['10em', '8rem']
   */
  offset?: Array<string | number>;
  /**
   * 形状
   * @default circle
   */
  shape?: 'circle' | 'round';
  /**
   * 当数值为 0 时，是否展示徽标
   * @default false
   */
  showZero?: boolean;
  /**
   * 尺寸
   * @default medium
   */
  size?: 'small' | 'medium';
}
