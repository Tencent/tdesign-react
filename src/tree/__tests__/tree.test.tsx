import React from 'react';
import { render, fireEvent, vi, mockTimeout, mockDelay } from '@test/utils';
import Tree from '../Tree';
import Button from '../../button';
import { TdTreeProps } from '../type';

// TODO
describe('Tree test', () => {
  // label 类型定义有问题
  const items = [
    {
      label: '第1一段',
      value: 1,
      children: [
        {
          label: '第二段',
          value: '1-1',
        },
        {
          label: '第二段',
          value: '1-2',
        },
      ],
    },
    {
      label: '第二段',
      value: 2,
    },
  ];
  async function renderTreeWithProps(props?: TdTreeProps) {
    const ref = {
      current: null,
    };
    const result = render(<Tree data={items} label={() => '自定义节点标签'} {...props} ref={ref} />);
    return {
      ...result,
      ref,
    };
  }

  test('empty Tree', async () => {
    const { container } = render(<Tree data={[]} />);
    await mockTimeout(() => {
      expect(container.querySelector('.t-tree')).toHaveTextContent('暂无数据');
    });
  });

  test('customize empty prop', async () => {
    const { container } = render(<Tree data={[]} empty="空数据（string）" />);
    await mockTimeout(() => {
      console.log('customize empty prop', container.querySelector('.t-tree'));
      expect(container.querySelector('.t-tree')).toHaveTextContent('空数据（string）');
    });
  });

  test('test trigger events', async () => {
    const data = [
      {
        label: 't-1',
        value: 1,
        children: [
          {
            label: 't-1-1',
            value: '1-1',
          },
        ],
      },
    ];
    const { container } = await renderTreeWithProps({ data, checkable: true });
    expect(container.querySelector('.t-tree')).not.toBeNull();
    // show node list
    await mockTimeout(() => {
      expect(container.querySelector('.t-tree__list').children.length).not.toBe(0);
    }, 100);
  });

  test('test Tree instance api', async () => {
    const data = [
      {
        label: 't-1',
        value: 1,
        children: [
          {
            label: 't-1-1',
            value: '1-1',
          },
          {
            label: 't-1-2',
            value: '1-2',
            children: [
              {
                label: 't-1-2-1',
                value: '1-2-1',
              },
              {
                label: 't-1-2-2',
                value: '1-2-2',
              },
              {
                label: 't-1-2-3',
                value: '1-2-3',
              },
            ],
          },
        ],
      },
      {
        label: 't-2',
        value: 2,
      },
    ];
    const { ref } = await renderTreeWithProps({ data, checkable: true });

    expect(ref.current.getIndex(1)).toBe(0);
    const node = ref.current.getItem(1);
    expect(node.value).toBe(1);
    expect(node.label).toBe('t-1');
    expect(ref.current.getItems(1).length).toEqual(6);
    expect(ref.current.getParent('1-2-1').value).toBe('1-2');
    expect(ref.current.getParents('1-2-1').length).toBe(2);
    expect(ref.current.getPath('1-2-1').length).toBe(3);
    ref.current.getPath('1-2-1');

    ref.current.insertBefore(1, { value: 'insert-before' });
    expect(ref.current.getItem('insert-before').value).toBe('insert-before');

    ref.current.insertAfter(2, { value: 'insert-after' });
    expect(ref.current.getItem('insert-after').value).toBe('insert-after');

    ref.current.appendTo(1, { value: '1-appendTo' });
    expect(ref.current.getItem('1-appendTo').value).toBe('1-appendTo');
    ref.current.remove('1-appendTo');
    expect(ref.current.getItem('1-appendTo')).toBeUndefined();

    ref.current.setItem(1, {
      label: 't-1-new',
      checked: true,
      actived: true,
      expanded: true,
      checkable: true,
    });
    await mockDelay(300);
    expect(ref.current.getItem(1).label).toBe('t-1-new');
    expect(ref.current.getItem(1).checked).toBe(true);
    expect(ref.current.getItem(1).expanded).toBe(true);
  });

  test('test render Tree operations, props.operations is a function', async () => {
    const fn1 = vi.fn();
    const renderOperations = (node) => (
      <>
        <Button style={{ marginLeft: '10px' }} size="small" variant="base" onClick={() => fn1(node)}>
          添加子节点
        </Button>
      </>
    );
    const { container } = await renderTreeWithProps({ operations: renderOperations });
    await mockDelay(300);
    expect(container.querySelector('.t-tree__operations')).not.toBeNull();
  });

  test('test render Tree operations function, props.operations is a reactNode', async () => {
    // operations 类型定义有些问题
    const renderOperations: any = (
      <>
        <Button style={{ marginLeft: '10px' }} size="small" variant="base">
          添加子节点
        </Button>
      </>
    );
    const { container } = await renderTreeWithProps({ operations: renderOperations });
    await mockDelay(300);
    expect(container.querySelector('.t-tree__operations')).not.toBeNull();
  });
  test('test render Tree operations error', async () => {
    const renderOperations: any = new Error('error operations');
    try {
      await renderTreeWithProps({ operations: renderOperations });
    } catch (err) {
      expect(err).toEqual(new Error('invalid type of operations'));
    }
  });

  it('when props.disableCheck is a function , it works fine', async () => {
    const onChangeFn1 = vi.fn();
    const { container } = await renderTreeWithProps({
      onChange: onChangeFn1,
      checkable: true,
      disableCheck: () => true,
    });

    await mockDelay(300);
    expect(container.querySelectorAll('.t-checkbox__input').length).toBe(2);
    fireEvent.click(container.querySelector('.t-checkbox__input'));
    await mockDelay(300);
    expect(container.querySelectorAll('.t-is-checked').length).toBe(0);
    expect(onChangeFn1).not.toHaveBeenCalled();
  });

  test('props.line', async () => {
    const data = [
      {
        label: '第1一段',
        value: 1,
        children: [
          {
            label: '第二段',
            value: '1-1',
            children: [
              {
                label: '第三段',
                value: '1-1-1',
              },
            ],
          },
        ],
      },
      {
        label: '第二段',
        value: 2,
        children: [
          {
            label: '2.1',
            value: '2-1',
            children: [
              {
                label: '2.1.1',
                value: '2-1-1',
              },
            ],
          },
        ],
      },
    ];
    it('wehen props.line is customized, it works fine', async () => {
      const { container } = await renderTreeWithProps({
        data,
        expandAll: true,
        line: <span className="custom-line">line</span>,
      });
      await mockDelay(300);
      expect(container.querySelectorAll('.custom-line').length).toBe(4);
    });

    it('wehen props.line is a customized function, it works fine', async () => {
      const line: any = (prams) => <span className="custom-line">{prams}</span>;
      const { container } = await renderTreeWithProps({
        data,
        expandAll: true,
        line,
      });
      await mockDelay(300);
      expect(container.querySelectorAll('.custom-line').length).toBe(4);
    });
  });
  test('when props.icon is a validElement, it works fine', async () => {
    const icon: any = <span>icon</span>;
    const { container } = await renderTreeWithProps({ icon });
    await mockDelay(300);
    expect(container.querySelectorAll('.t-tree__icon').length).toBe(2);
  });

  it('when props.icon is a validElement, it works fine', async () => {
    const icon: any = <span>icon</span>;
    const { container } = await renderTreeWithProps({ icon });
    await mockDelay(300);
    expect(container.querySelectorAll('.t-tree__icon').length).toBe(2);
  });
  it('when props.icon is a function, it works fine', async () => {
    const icon: any = () => <span>icon</span>;
    const { container } = await renderTreeWithProps({ icon });
    await mockDelay(300);
    expect(container.querySelectorAll('.t-tree__icon').length).toBe(2);
  });

  // TODO: icon error can't be caught by vitest
  // it('test render Tree icon error', async () => {
  //   const icon: any = new Error('error icon');
  //   try {
  //     renderTreeWithProps({ icon });
  //   } catch (err) {
  //     expect(err).toEqual(new Error('invalid type of icon'));
  //   }
  // });

  test('TreeNodeState.loading works fine', async () => {
    const data = [
      {
        label: '第1一段',
        value: 1,
        loading: true,
        expanded: true,
        children: [
          {
            label: '第二段',
            value: '1-1',
          },
        ],
      },
      {
        label: '第二段',
        value: 2,
      },
    ];
    const { container } = await renderTreeWithProps({ data });
    await mockDelay(300);
    expect(container.querySelectorAll('.t-loading').length).toBe(1);
  });
});
