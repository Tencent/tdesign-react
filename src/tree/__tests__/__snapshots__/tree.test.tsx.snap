// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activable.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 120px;"
          >
            <label>
              ËäÇÁÇπÂèØÈ´ò‰∫Æ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 120px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 120px;"
          >
            <label>
              ËäÇÁÇπÂèØÂ§öÈÄâÈ´ò‰∫Æ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 120px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 120px;"
          >
            <label>
              Êï¥‰∏™ËäÇÁÇπÂèØÁÇπÂáª
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 120px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`base.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`checkable.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 100px;"
          >
            <label>
              ÂèØÈÄâ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 100px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 100px;"
          >
            <label>
              ‰∏•Ê†ºÊ®°Âºè
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 100px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item t-form-item__valueMode"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 100px;"
          >
            <label>
              ÈÄâ‰∏≠ÂÄºÊ®°Âºè
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 100px;"
          >
            <div
              class="t-form__controls-content"
            >
              <div
                class="t-radio-group t-size-m t-radio-group__outline"
              >
                <label
                  class="t-radio t-is-checked"
                >
                  <input
                    checked=""
                    class="t-radio__former"
                    readonly=""
                    type="radio"
                    value="onlyLeaf"
                  />
                  <span
                    class="t-radio__input"
                  />
                  <span
                    class="t-radio__label"
                  >
                    onlyLeaf
                  </span>
                </label>
                <label
                  class="t-radio"
                >
                  <input
                    class="t-radio__former"
                    readonly=""
                    type="radio"
                    value="parentFirst"
                  />
                  <span
                    class="t-radio__input"
                  />
                  <span
                    class="t-radio__label"
                  >
                    parentFirst
                  </span>
                </label>
                <label
                  class="t-radio"
                >
                  <input
                    class="t-radio__former"
                    readonly=""
                    type="radio"
                    value="all"
                  />
                  <span
                    class="t-radio__input"
                  />
                  <span
                    class="t-radio__label"
                  >
                    all
                  </span>
                </label>
              </div>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="t-tree t-tree--hoverable t-tree--checkable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`controlled.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        checked:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value="1.1.1.1, 1.1.1.2"
        />
      </div>
    </div>
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        expanded:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value="1, 1.1, 1.1.1, 2"
        />
      </div>
    </div>
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        actived:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value="2"
        />
      </div>
    </div>
    <div
      class="t-tree t-tree--checkable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`disabled.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 120px;"
          >
            <label>
              ÊòØÂê¶Á¶ÅÁî®
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 120px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="t-tree t-is-disabled t-tree--hoverable t-tree--checkable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`empty.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-tree t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
    <br />
    <div
      class="t-tree t-tree--transition"
    >
      üòä Á©∫Êï∞ÊçÆÔºàstringÔºâ
    </div>
    <br />
    <div
      class="t-tree t-tree--transition"
    >
      <div>
        üòä Á©∫Êï∞ÊçÆÔºà empty props Ôºâ
      </div>
    </div>
    <br />
  </div>
</DocumentFragment>
`;

exports[`expand-all.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 150px;"
          >
            <label>
              ÂàáÊç¢Êï∞ÊçÆ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 150px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 150px;"
          >
            <label>
              Êèê‰æõ hover Áä∂ÊÄÅ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 150px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 150px;"
          >
            <label>
              Â±ïÂºÄÂä®Áîª
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 150px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`expand-level.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`expand-mutex.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 120px;"
          >
            <label>
              ‰∫íÊñ•Â±ïÂºÄ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 120px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 120px;"
          >
            <label>
              Êï¥‰∏™ËäÇÁÇπÂèØÁÇπÂáª
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 120px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="t-tree t-tree--hoverable t-tree--transition t-tree--block-node"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`filter.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        filter:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value=""
        />
      </div>
    </div>
    <div
      class="t-tree t-tree--hoverable t-tree--transition t-tree--block-node"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`icon.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <h3>
      render 1:
    </h3>
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
    <h3>
      render 2:
    </h3>
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`label.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-tree t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`lazy.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`line.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-line"
  >
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 100px;"
          >
            <label>
              ÊòæÁ§∫ËøûÁ∫ø
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 100px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 100px;"
          >
            <label>
              ÊòæÁ§∫ÂõæÊ†á
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 100px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="t-tree t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
    <h3>
      render
    </h3>
    <div
      class="t-tree t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`load.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;

exports[`operations.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <h3
      class="title"
    >
      render:
    </h3>
    <div
      class="t-tree t-tree--hoverable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
    <h3
      class="title"
    >
      api:
    </h3>
    <div
      class="operations"
    >
      <form
        class="t-form"
      >
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 200px;"
          >
            <label>
              ÊèíÂÖ•ËäÇÁÇπ‰ΩøÁî®È´ò‰∫ÆËäÇÁÇπ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 200px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
        <div
          class="t-form__item"
        >
          <div
            class="t-form__label t-form__label--right"
            style="width: 200px;"
          >
            <label>
              Â≠êËäÇÁÇπÂ±ïÂºÄËß¶ÂèëÁà∂ËäÇÁÇπÂ±ïÂºÄ
            </label>
          </div>
          <div
            class="t-form__controls"
            style="margin-left: 200px;"
          >
            <div
              class="t-form__controls-content"
            >
              <button
                class="t-switch t-is-checked"
                role="switch"
                type="button"
              >
                <span
                  class="t-switch__handle"
                />
                <div
                  class="t-switch__content"
                />
              </button>
            </div>
          </div>
        </div>
      </form>
    </div>
    <div
      class="operations"
    >
      <div
        class="t-addon t-addon--prepend"
      >
        <span
          class="t-addon__prepend"
        >
          filter:
        </span>
        <div
          class="t-input t-input__inner"
        >
          <input
            value=""
          />
        </div>
      </div>
    </div>
    <div
      class="t-tree t-tree--hoverable t-tree--checkable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
    <h3
      class="title"
    >
      api:
    </h3>
    <div
      class="operations"
    >
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑Âèñ value ‰∏∫ 'node1' ÁöÑÂçï‰∏™ËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÊâÄÊúâËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÈ´ò‰∫ÆËäÇÁÇπÁöÑÊâÄÊúâÂ≠êËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÊâÄÊúâÈ´ò‰∫ÆËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÈ´ò‰∫ÆËäÇÁÇπ‰∏ãÁöÑÈÄâ‰∏≠ËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          ÊèíÂÖ•‰∏Ä‰∏™Ê†πËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÈ´ò‰∫ÆËäÇÁÇπÁöÑÁà∂ËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÈ´ò‰∫ÆËäÇÁÇπÁöÑÊâÄÊúâÁà∂ËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÈ´ò‰∫ÆËäÇÁÇπÂú®Â≠êËäÇÁÇπ‰∏≠ÁöÑ‰ΩçÁΩÆ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          ÈÄâ‰∏≠È´ò‰∫ÆËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Â±ïÂºÄÈ´ò‰∫ÆËäÇÁÇπ
        </span>
      </button>
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="margin: 0px 10px 10px 0px; position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          Ëé∑ÂèñÈ´ò‰∫ÆËäÇÁÇπ‰∏éÂÖ∂Â≠êËäÇÁÇπÁöÑÊï∞ÊçÆ
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`state.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-state"
  >
    <h3
      class="title"
    >
      state:
    </h3>
    <div
      class="t-tree t-tree--hoverable t-tree--checkable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
    <h3
      class="title"
    >
      api:
    </h3>
    <div
      class="operations"
    >
      <button
        class="t-button t-button--theme-primary t-button--variant-base"
        style="position: relative;"
        type="button"
      >
        <span
          class="t-button__text"
        >
          ÊèíÂÖ•‰∏Ä‰∏™Ê†πËäÇÁÇπ
        </span>
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`sync.jsx 1`] = `
<DocumentFragment>
  <div
    class="tdesign-tree-base"
  >
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        checked:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value="{1.1.1.1}, {1.1.1.2}"
        />
      </div>
    </div>
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        expanded:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value="{1}, {1.1}, {1.1.1}, {2}"
        />
      </div>
    </div>
    <div
      class="t-addon t-addon--prepend"
    >
      <span
        class="t-addon__prepend"
      >
        actived:
      </span>
      <div
        class="t-input t-input__inner"
      >
        <input
          value="{2}"
        />
      </div>
    </div>
    <div
      class="t-tree t-tree--checkable t-tree--transition"
    >
      ÊöÇÊó†Êï∞ÊçÆ
    </div>
  </div>
</DocumentFragment>
`;
